python/lsst/obs/auxTel/ingest.py

```diff
@@ -0,0 +1,129 @@
+
+    def getInfo(self, filename):
+        """ Get the basename and other data which is only available from the filename/path.
+
+        This seems fragile, but this is how the teststand data will *always* be written out, 
+        as the software has been "frozen" as they are now in production mode.
+
+        Parameters
+        ----------
+        filename : `str`
+            The filename
+
+        Returns
+        -------
+        phuInfo : `dict`
+            Dictionary containing the header keys defined in the ingest config from the primary HDU
+        infoList : `list`
+            A list of dictionaries containing the phuInfo(s) for the various extensions in MEF files
+        """
+        phuInfo, infoList = ParseTask.getInfo(self, filename)
+
+        pathname, basename = os.path.split(filename)
+        basename = re.sub(r"\.(%s)$" % "|".join(EXTENSIONS), "", basename)
+        phuInfo['basename'] = basename
+
+        # Now pull the acq type & jobID from the path (no, they're not in the header)
+        # the acq type is the type of test, eg flat/fe55/darks etc
+        # jobID is the test number, and corresponds to database entries in the eTraveller/cameraTestDB
+        pathComponents = pathname.split("/")
+        if len(pathComponents) < 0:
+            raise RuntimeError("Path %s is too short to deduce raftID" % pathname)
+        sensorId, acquisitionType, testVersion, runId = pathComponents[-4:]
+
+        if phuInfo['lsstSerial'] != sensorId:
+            raise RuntimeError("Serial mismatch: the header says %s but found %s from"
+                               "path %s"%(phuInfo['lsstSerial'], sensorId, pathname))
+        phuInfo['run'] = runId  # NOT in the header
+        phuInfo['field'] = acquisitionType # NOT in the header
+        phuInfo['ccd'] = 0  # There is only one ccd in this camera
+
+        return phuInfo, infoList
+
+    def translate_wavelength(self, md):
+        """Translate wavelength provided by teststand readout.
+
+        The teststand driving script asks for a wavelength, and then reads the value back to ensure that
+        the correct position was moved to. This number is therefore read back with sub-nm precision.
+        Typically the position is within 0.005nm of the desired position, so we warn if it's not very
+        close to an integer value.
+
+        Future users should be aware that the HIERARCH MONOCH-WAVELENG key is NOT the requested value, and
+        therefore cannot be used as a cross-check that the wavelength was close to the one requested.
+        The only record of the wavelength that was set is in the original filename.
+
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList or PropertySet`
+            image metadata
+
+        Returns
+        -------
+        wavelength : `int`
+            The recorded wavelength as an int
+        """
+        raw_wl = md.get("MONOWL")
+        wl = int(round(raw_wl))
```

Comment:
Doesn't `round` return an `int`?  It does in python3, but maybe not in python2.

Reply:
You're right, but it doesn't do much harm to ensure that it's an int so that it works for those still in py2 land does it. Do you think this needs removing?